src/natcap/invest/recreation/recmodel_server.py

```diff
@@ -45,6 +45,36 @@
 LOGGER = logging.getLogger('natcap.invest.recreation.recmodel_server')
 
 
+def _numpy_dumps(numpy_array):
+    """Safely pickle numpy array to string.
+    Args:
+        numpy_array (numpy.ndarray): arbitrary numpy array.
+    Returns:
+        A string representation of the array that can be loaded using
+        `numpy_loads.
+    """
+    with BytesIO() as file_stream:
+        numpy.save(file_stream, numpy_array, allow_pickle=False)
+        return file_stream.getvalue()
+
+
+def _numpy_loads(queue_string):
+    """Safely unpickle numpy array to string.
+    Items in our queue may either be a pickled numpy array string, or the
+    string 'STOP'. If 'STOP', don't try to load as a numpy array.
+    Args:
+        queue_string (str): binary string representing a pickled
+            numpy array.
+    Returns:
+        A numpy representation of ``binary_numpy_string``.
+    """
+    if queue_string == 'STOP':
+        return queue_string
```

Comment:
I mention below that I think it's a cleaner solution to explicitly check the sentinel before trying to decode it, if you agree, you can simplify this part.

Reply:
Agree it is cleaner to keep this out of the `_numpy_loads` function!